### PR TITLE
feat: ceramic, didprovider, idw 2 support

### PR DESCRIPTION
**Pull request by zachferland**
npm run start:example to try

~~will tag for review once i refactor some paths, but can be released to start using anywhere~~ 

need to change account link message in blockchain utils or allow message pass

### CODE REVIEW

File: src/authProvider/ethereumAuthProvider.js [hunk 1]
```diff
@@ -1,21 +1,32 @@
 import { createLink, authenticate } from '3id-blockchain-utils'
 import AbstractAuthProvider from './abstractAuthProvider'
+import { AccountID } from "caip"
+
+const chainId = "eip155:1"
```
Comment (oed) [inline src/authProvider/ethereumAuthProvider.js:5]: Can't we just get this from the provider?

**Review by oed — APPROVED** [review]
This looks good 👍

File: iframe/assets/logo.svg [hunk 2]
```diff
@@ -1,7 +1,18 @@
-<svg width="290px" height="289px" viewBox="0 0 290 289" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
```
Comment (PaulLeCam) [inline iframe/assets/logo.svg:1]: Optimized version using SVGO (https://jakearchibald.github.io/svgomg/):

```xml
<svg width="334" height="334" xmlns="http://www.w3.org/2000/svg">
  <defs/>
  <g fill="none" fill-rule="evenodd">
    <path d="M171.57 1.044a10.4 10.4 0 014.835 4.831c19.291 39.83 39.315 70.124 60.071 90.88 20.576 20.576 50.64 40.55 90.195 59.921l1.226.6c5.164 2.514 7.311 8.74 4.796 13.903a10.4 10.4 0 01-4.831 4.814c-39.358 18.984-69.505 38.806-90.444 59.465-20.884 20.607-41.089 51.006-60.613 91.197l-.592 1.22a10.4 10.4 0 01-9.173 5.877V.006a10.358 10.358 0 014.53 1.038z" fill="#FF3600"/>
    <path d="M168.806 82.266a4 4 0 011.901 1.9c9.974 21.016 20.342 36.905 31.103 47.667 10.603 10.602 26.245 20.885 46.927 30.849l1.004.481a4 4 0 01-.014 7.227c-20.77 9.814-36.586 20.079-47.446 30.794-10.752 10.61-21.143 26.41-31.175 47.4l-.477 1.003a4 4 0 01-3.536 2.288l-.001-169.995c.575 0 1.16.123 1.714.386zm-86.514 82.78v3.69a3.995 3.995 0 01-.026-3.641l.026-.049z" fill="#FFB08A"/>
    <g>
      <path d="M162.67 1.038a10.4 10.4 0 00-4.835 4.831c-19.291 39.83-39.315 70.124-60.071 90.88-20.576 20.576-50.64 40.55-90.195 59.921l-1.226.6c-5.164 2.514-7.311 8.74-4.796 13.903a10.4 10.4 0 004.831 4.814c39.358 18.984 69.505 38.806 90.444 59.465 20.884 20.607 41.089 51.006 60.613 91.197l.592 1.22a10.4 10.4 0 009.173 5.877V0a10.358 10.358 0 00-4.53 1.038z" fill="#FFB08A"/>
      <path d="M165.437 82.26a4 4 0 00-1.901 1.9c-9.974 21.016-20.342 36.905-31.104 47.667-10.602 10.602-26.244 20.885-46.926 30.848l-1.005.482a4 4 0 00.015 7.227c20.77 9.814 36.585 20.078 47.445 30.794 10.753 10.61 21.144 26.41 31.175 47.4l.478 1.003a4 4 0 003.536 2.288V81.874c-.574 0-1.159.123-1.713.386zm86.514 82.78v3.69c.57-1.1.624-2.45.025-3.642l-.025-.048z" fill="#FF3600"/>
    </g>
  </g>
</svg>
```

File: iframe/css/style.scss [hunk 3]
```diff
@@ -340,7 +347,7 @@ button {
 
 .actions {
   display: flex;
-  flex-direction: column;
+  // flex-direction: column;
```
Comment (PaulLeCam) [inline iframe/css/style.scss:350]: To remove maybe if not used?

File: src/connectService.js [hunk 4]
```diff
@@ -6,15 +6,16 @@ const sha256 = require('js-sha256').sha256
 import { publishedDefinitions, publishedSchemas } from '@ceramicstudio/idx-tools'
```
Comment (PaulLeCam) [inline src/connectService.js:6]: It's better to use [idx-constants](https://github.com/ceramicstudio/js-idx-constants) for runtime code if possible as `idx-tools` comes with extra code and dependencies

Comment (zachferland) [reply src/connectService.js:6]: ah perfect, yeah will use that

File: iframe/css/style.scss (hunk 3 above)
Comment (zachferland) [reply iframe/css/style.scss:350]: 👍

File: iframe/assets/logo.svg (hunk 2 above)
Comment (zachferland) [reply iframe/assets/logo.svg:1]: nice, will pull that in, thanks!

File: src/authProvider/ethereumAuthProvider.js (hunk 1 above)
Comment (zachferland) [reply src/authProvider/ethereumAuthProvider.js:5]: in general? or from eth providers?

Comment (oed) [reply src/authProvider/ethereumAuthProvider.js:5]: from an eth provider you can  get the number from ` net_version`. https://eth.wiki/json-rpc/API

You would still need to prepend `eip155:`

**Comment by zachferland** [comment]
Merged into develop-ceramic instead

@PaulLeCam available at 3id-connect@next, 3idconnect.org, and no docs yet, but examples shows how 3id-connect is used

**Comment by PaulLeCam** [comment]
Great, thanks!

**Pull request merged**